[Caffe2] Add float batch box cox SVE128 implementation

Nicoshev opened the pull request:
Introduce SVE128 SIMD batch box-cox computation.

We've seen about 65% throughput improvement.

Privacy Context Container: L1196524

This is a no-op from OSS point of view, therefore it could be landed without tests (see precedence set by https://github.com/pytorch/pytorch/pull/143627), but we should delete those at some point


cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

malfet commented on caffe2/perfkernels/batch_box_cox_Sve128.cc:
```diff
@@ -0,0 +1,241 @@
+  const float32x4_t log_tab_7 = vdupq_n_f32(4.58445882797f);
+  const float32x4_t log_tab_8 = vdupq_n_f32(0.0141278216615f);
+
+  float32x4_t A = vmlaq_f32(log_tab_1, log_tab_5, x);
```
Please use `auto` to avoid potential casting errors (both clang and gcc will silently typecast NEON return to incorrect vectorized type, but will error out if its passed as an argument)
```suggestion
  const auto A = vmlaq_f32(log_tab_1, log_tab_5, x);
```

malfet reviewed (changes requested):
- Please used `auto` to avoid some hard to spot casting errors
- Use `C10_LIKELY`/`C10_UNLIKELY` instead of `__builtin_expect`
- Add new unittest or mention in PR which test validates this one

Nicoshev committed:
[Caffe2] Add float batch box cox SVE128 implementation (#159778)

Summary:

We are introducing SVE128 perfkernels

As first translation, we are implementing float32 batch box cox for SVE128

Test Plan:
Sigrid Predictor canary

Rollback Plan:

Differential Revision:
D78994871

Privacy Context Container: L1196524
